enterprise/internal/own/background/analytics.go

```diff
@@ -18,33 +20,44 @@ import (
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
-func handleAnalytics(ctx context.Context, logger log.Logger, repoId api.RepoID, db database.DB) error {
+func handleAnalytics(ctx context.Context, lgr log.Logger, repoId api.RepoID, db database.DB, subRepoPermsCache *rcache.Cache) error {
 	// 🚨 SECURITY: we use the internal actor because the background indexer is not associated with any user,
 	// and needs to see all repos and files.
 	internalCtx := actor.WithInternalActor(ctx)
-	indexer := newAnalyticsIndexer(gitserver.NewClient(), db)
-	err := indexer.indexRepo(internalCtx, repoId)
+	indexer := newAnalyticsIndexer(gitserver.NewClient(), db, subRepoPermsCache, lgr)
+	err := indexer.indexRepo(internalCtx, repoId, authz.DefaultSubRepoPermsChecker)
 	if err != nil {
-		logger.Error("own analytics indexing failure", log.String("msg", err.Error()))
+		lgr.Error("own analytics indexing failure", log.String("msg", err.Error()))
 	}
 	return err
 }
 
 type analyticsIndexer struct {
-	client gitserver.Client
-	db     database.DB
+	client            gitserver.Client
+	db                database.DB
+	logger            log.Logger
+	subRepoPermsCache *rcache.Cache
 }
 
-func newAnalyticsIndexer(client gitserver.Client, db database.DB) *analyticsIndexer {
-	return &analyticsIndexer{client: client, db: db}
+func newAnalyticsIndexer(client gitserver.Client, db database.DB, subRepoPermsCache *rcache.Cache, lgr log.Logger) *analyticsIndexer {
+	return &analyticsIndexer{client: client, db: db, subRepoPermsCache: subRepoPermsCache, logger: lgr}
 }
 
 var ownAnalyticsFilesCounter = promauto.NewCounter(prometheus.CounterOpts{
 	Namespace: "src",
 	Name:      "own_analytics_files_indexed_total",
 })
 
-func (r *analyticsIndexer) indexRepo(ctx context.Context, repoId api.RepoID) error {
+func (r *analyticsIndexer) indexRepo(ctx context.Context, repoId api.RepoID, checker authz.SubRepoPermissionChecker) error {
+	// If the repo has sub-repo perms enabled, skip indexing
+	isSubRepoPermsRepo, err := isSubRepoPermsRepo(ctx, repoId, r.subRepoPermsCache, checker)
+	if err != nil {
+		return err
```

Comment:
Although this error is unlikely to happen, but I think it should be non-retryable?
```suggestion
		return errcode.MakeNonRetryable(err)
```